File: fixture-app-python/format_test.py
import json

from format import format_order


def test_format_order_rejects_negative_amount():
    status, body = format_order(-5)
    assert status == 400
    payload = json.loads(body)
    assert "error" in payload


def test_format_order_accepts_positive_amount():
    status, body = format_order(42)
    assert status == 200
    payload = json.loads(body)
    assert payload == {"amount": 42, "currency": "USD"}


def test_format_order_accepts_zero_amount():
    status, body = format_order(0)
    assert status == 200
    payload = json.loads(body)
    assert payload == {"amount": 0, "currency": "USD"}

File: fixture-app-node/format.test.js
const { formatOrder } = require("./format");

describe("formatOrder", () => {
  test("rejects negative amounts with 400", () => {
    const { status, body } = formatOrder(-5);
    expect(status).toBe(400);
    const parsed = JSON.parse(body);
    expect(parsed.amount).toBeUndefined();
  });

  test("accepts non-negative amounts with 200 and echoes them back", () => {
    const { status, body } = formatOrder(42);
    expect(status).toBe(200);
    const parsed = JSON.parse(body);
    expect(parsed.amount).toBe(42);
    expect(parsed.currency).toBe("USD");
  });

  test("accepts zero amount with 200", () => {
    const { status, body } = formatOrder(0);
    expect(status).toBe(200);
    const parsed = JSON.parse(body);
    expect(parsed.amount).toBe(0);
  });
});
